projects/storefrontlib/src/shared/components/form/form-errors/form-errors.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+/**
+ * This component renders form errors.
+ */
+@Component({
+  selector: 'cx-form-errors',
+  templateUrl: './form-errors.component.html',
+})
+export class FormErrorComponent {
+  @Input() control: FormControl;
+
+  /**
+   * Checks if the control is invalid (contains any validation errors).
+   * Also checks if the control was already touched (either with click or focus).
+   * @returns boolean
+   */
+  shouldShowErrors(): boolean {
+    return this.control.invalid && (this.control.dirty || this.control.touched);
```

Comment:
I'd still prefer to leave this logic up to the css. In CSS we can decide (per formcontrol) to show certain errors . This gives more flexibility.

Reply:
this method was removed altogether

Reply:
`form-errors` element's visibility is now CSS-driven, but it's making the component less 'universal', because now we have to use `cx-form-errors` right after the input (`input + cx-form-errors` in CSS)